packages/atclient/CMakeLists.txt

```diff
@@ -124,6 +128,13 @@ if(NOT ESP_PLATFORM)
       $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   )
 
+  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+    target_include_directories(
+      ${PROJECT_NAME}
+      PUBLIC $<BUILD_INTERFACE:${ATCLIENT_PROTECTED_INCLUDE_DIR}>
+    )
+  endif()
+
```

Comment:
Include the protected dir if we are in debug mode - compile_commands and tests should be generated / built in debug mode.

Customer facing package builds should be generated in release mode